File: reynolds-delicious-restaurant/src/com/reynolds/decious/restaurant/entities/MenuItem.java
package com.reynolds.decious.restaurant.entities;

import java.util.Objects;

public class MenuItem {
    private Long id;
    private Menu menu;
    private FinalProduct finalProduct;
    private Double price;

    public MenuItem() {
    }

    public MenuItem(Long id) {
        this.id = id;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Menu getMenu() {
        return menu;
    }

    public void setMenu(Menu menu) {
        this.menu = menu;
    }

    public FinalProduct getFinalProduct() {
        return finalProduct;
    }

    public void setFinalProduct(FinalProduct finalProduct) {
        this.finalProduct = finalProduct;
    }

    public Double getPrice() {
        return price;
    }

    public void setPrice(Double price) {
        this.price = price;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        MenuItem menuItem = (MenuItem) o;
        return Objects.equals(id, menuItem.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}

File: millord-restaurant-data/src/entities/MenuItem.ts
import { ModuleName } from 'typescript';
import { Menu } from './Menu'
import { FinalProduct } from './FinalProduct'

import { Md5 } from 'ts-md5/dist/md5'

export class MenuItem {
<<<<<<< HEAD
  private _id?: string;
  private _menu?: Menu;
  private _finalProduct?: FinalProduct
  private _price?: number;

  constructor() { }

  get id(): string | undefined {
    return this._id;
  }

  set id(value: string | undefined) {
    this._id = value
  }

  get menu(): Menu | undefined {
    return this._menu;
  }

  set menu(value: Menu | undefined) {
    this._menu = value;
  }

  get finalProduct(): FinalProduct | undefined {
    return this._finalProduct
  }

  set finalProduct(value: FinalProduct | undefined) {
    this._finalProduct = value
  }

  get price(): number | undefined {
    return this._price
  }

  set price(value: number | undefined) {
    this._price = value
=======
  private id?: string;
  private menu?: Menu;
  private finalProduct?: FinalProduct
  private price?: number;

  constructor() {

  }


  public getId(): string | undefined {
    return this.id;
  }

  public setId(id: string): void {
    this.id = id;
  }

  public getMenu(): Menu | undefined {
    return this.menu;
  }

  public setMenu(menu: Menu): void {
    this.menu = menu;
  }

  public getFinalProduct(): FinalProduct | undefined {
    return this.finalProduct;
  }

  public setFinalProduct(finalProduct: FinalProduct): void {
    this.finalProduct = finalProduct;
  }

  public getPrice(): number | undefined {
    return this.price;
  }

  public setPrice(price: number): void {
    this.price = price;
  }


  public hashCode() {
    return Md5.hashStr(this.id)
>>>>>>> e5e309a229e3d2976167e70c766c15802cac1c9b
  }




}